### PR TITLE
Implement Sign Up flow on the server

### PR DESCRIPTION
**Pull request by vikachuu**

### CODE REVIEW

File: backend/src/common/models/user.ts [hunk 1]
```diff
@@ -1,3 +1,23 @@
...
+  email: string,
+  password: string,
+  displayName: string,
+  imageUrl: string,
+  title: string
+}
+
+export interface IUserResponse extends IUser {
+  token: string,
+}
```
Comment (AlikRakhmonov) [inline backend/src/common/models/user.ts:23]: create folder common/models/user and create a file for every interface there. Name files by their interface names

File: backend/src/common/models/user.ts [hunk 2]
```diff
@@ -1,3 +1,23 @@
...
+
+export interface IUser {
+  id: string,
+  fullName: string,
+  email: string,
+  password: string,
+  displayName: string,
+  imageUrl: string,
+  title: string
```
Comment (AlikRakhmonov) [inline backend/src/common/models/user.ts:18]: use semicolons in interfaces

File: backend/src/data/repositories/userRepository.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { EntityRepository, Repository } from 'typeorm';
+import { User } from '../entities/User/UserEntity';
+import { IRegisterUser, IUser } from '../../common/models/user';
+
+@EntityRepository(User)
+class UserRepository extends Repository<User> {
+  async addUser(data: IRegisterUser): Promise<IUser> {
+    const user = this.create({
+      ...data,
+      displayName: data.fullName
```
Comment (AlikRakhmonov) [inline backend/src/data/repositories/userRepository.ts:10]: map display name in service

File: backend/src/data/repositories/userRepository.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { EntityRepository, Repository } from 'typeorm';
+import { User } from '../entities/User/UserEntity';
+import { IRegisterUser, IUser } from '../../common/models/user';
+
+@EntityRepository(User)
+class UserRepository extends Repository<User> {
+  async addUser(data: IRegisterUser): Promise<IUser> {
```
Comment (AlikRakhmonov) [inline backend/src/data/repositories/userRepository.ts:7]: why does it return IUser? user has User type.

File: backend/src/services/authService.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import { encrypt } from '../common/utils/encryptHelper';
+import { signUpResponseMapper } from '../common/utils/userHelper';
+
+export const register = async ({ password, ...userData }: IRegisterUser) => {
+  const newUser = await getCustomRepository(UserRepository).addUser({
+    ...userData,
+    password: await encrypt(password)
+  });
+
+  return signUpResponseMapper(newUser);
```
Comment (AlikRakhmonov) [inline backend/src/services/authService.ts:15]: ensure that you do not return password

File: backend/src/common/models/user.ts [hunk 6]
```diff
@@ -1,3 +1,23 @@
...
+  fullName: string;
+  email: string;
+  password: string;
+}
+
+export interface IUser {
+  id: string,
+  fullName: string,
+  email: string,
+  password: string,
```
Comment (AlikRakhmonov) [inline backend/src/common/models/user.ts:15]: do not return password. It's unsafe

File: backend/src/common/utils/userHelper.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import { createToken } from './tokenHelper';
+import { IUser } from '../models/user/IUser';
+import { IUserResponse } from '../models/user/IUserResponse';
+import { IRegisterUser } from '../models/user/IRegisterUser';
+import { ICreateUser } from '../models/user/ICreateUser';
+
+export const signUpResponseMapper = (user: IUser): IUserResponse => {
+  const responseUser = user;
+  delete responseUser.password;
```
Comment (vikachuu) [inline backend/src/common/utils/userHelper.ts:9]: maybe not best solution, but `const { password, ...responseUser } = user` causes linter error.

**Comment by vikachuu** [comment]
@AlikRakhmonov All comments are fixed. Please, review it again.

**Pull request merged**